mobify/progressive-web-scaffold
# Re-enable Lighthouse test on CI

Run Lighthouse tests on CircleCI again, with some changes

## Changes
- Move steps (starting server, starting proxy, running the test) into their own scripts

## How to test-drive this PR
- Checkout this branch
- `cd web`
- `npm i`
- `npm run test:pwa-prod` to test against the production site (https://www.merlinspotions.com)
- `npm run dev` in one tab and then `sudo npm run test:pwa-local` in another tab to test against the local build with Preview (via proxy) 

or 
- Rebuild https://circleci.com/gh/mobify/progressive-web-scaffold/tree/lighthouse-commands

The Lighthouse report will be opened in your default browser or can be found under the Artifacts tab on Circle.

*Notes* 
- We're making use of the `--allow-insecure-localhost` flag in order to test a service worker in dev environment (localhost). This has historically been buggy and was only recently fixed in Chrome 57 (still in beta) https://bugs.chromium.org/p/chromium/issues/detail?id=561820. If the Lighthouse score is lower than expected, it could be because the service worker is not being fetched due to the dev environment.
